### PR TITLE
add animation when button is clicked and fix title in button

### PR DESCRIPTION
**Pull request by AbbiePetcht**
- add animation when button is clicked
- move title to button.component so that the title can change through property

### CODE REVIEW

File: src/sql/parts/modelComponents/toolbarLayout.css [hunk 1]
```diff
@@ -62,4 +62,12 @@
 	margin-right: 0.3em;
 	background-position: 50% 50%;
 
-}
\ No newline at end of file
+}
```
Comment (anthonydresser) [inline src/sql/parts/modelComponents/toolbarLayout.css]: What is this animation and why do we need it?

Comment (kevcunnane) [reply src/sql/parts/modelComponents/toolbarLayout.css]: It matches the behavior of the built-in SQL toolbar. Any extension will now behave the same as our standard toolbar so it's consistent.

File: src/sql/base/browser/ui/button/button.ts [hunk 2]
```diff
@@ -23,6 +23,18 @@ export class Button extends vsButton {
 			this.$el.style('outline-color', this.buttonFocusOutline ? this.buttonFocusOutline.toString() : null);
 			this.$el.style('outline-width', '1px');
 		});
+
+		this.$el.on(DOM.EventType.MOUSE_DOWN, (e) => {
+			const mouseEvent = e as MouseEvent;
+			if (!this.$el.hasClass('disabled') && mouseEvent.button === 0) {
```
Comment (kevcunnane) [inline src/sql/base/browser/ui/button/button.ts]: Does this have impact outside of the extensibility scenarios? Can this impact any core scenario?

File: src/sql/parts/modelComponents/button.component.ts [hunk 3]
```diff
@@ -82,6 +82,9 @@ export default class ButtonComponent extends ComponentWithIconBase implements IC
 						args: e
 					});
 				}
+
+				// unfocus the button when it is already clicked
+				this._button.blur();
```
Comment (kevcunnane) [inline src/sql/parts/modelComponents/button.component.ts]: Can we test the dialog & wizard scenarios to verify this doesn't negatively impact them?

File: src/sql/parts/modelComponents/toolbarContainer.component.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export class ToolbarItem {
 	template: `
 		<div #container *ngIf="items" [class]="toolbarClass" >
 			<ng-container *ngFor="let item of items">
-			<div class="modelview-toolbar-item" [title]="getItemTitle(item)" [style.paddingTop]="paddingTop" tabindex="0">
```
Comment (kevcunnane) [inline src/sql/parts/modelComponents/toolbarContainer.component.ts]: Why get rid of tabIndex? I think it was likely added for accessibility - can you check with other folks on this?

Comment (AbbiePetcht) [reply src/sql/parts/modelComponents/toolbarContainer.component.ts]: Each component is a control and it is tabable. By doing this, you have 2 layers of tab

---
In reply to: [216546738](https://github.com/Microsoft/sqlopsstudio/pull/2488#discussion_r216546738) [](ancestors = 216546738)

File: src/sql/base/browser/ui/button/button.ts (hunk 2 above)
Comment (AbbiePetcht) [reply src/sql/base/browser/ui/button/button.ts]: It shouldn't. All it does is adding/removing "active" class in the dom. VS button doesn't use active class for anything

---
In reply to: [216546546](https://github.com/Microsoft/sqlopsstudio/pull/2488#discussion_r216546546) [](ancestors = 216546546)

File: src/sql/parts/modelComponents/button.component.ts (hunk 3 above)
Comment (AbbiePetcht) [reply src/sql/parts/modelComponents/button.component.ts]: I removed blur(). The correct scenario is to focus on the new UI (ie. focus on new cell, terminal window)

---
In reply to: [216546651](https://github.com/Microsoft/sqlopsstudio/pull/2488#discussion_r216546651) [](ancestors = 216546651)

**Pull request merged**